packages/compass-collection/src/stores/collection.ts

```diff
@@ -0,0 +1,113 @@
+import type AppRegistry from 'hadron-app-registry';
+import type { DataService } from 'mongodb-data-service';
+import { createStore, applyMiddleware } from 'redux';
+import thunk from 'redux-thunk';
+import reducer, {
+  collectionStatsFetched,
+  selectTab,
+} from '../modules/collection-tab';
+import type Collection from 'mongodb-collection-model';
+import toNs from 'mongodb-ns';
+import type { MongoDBInstance } from 'mongodb-instance-model';
+import type { CollectionMetadata } from 'mongodb-collection-model';
+
+export function configureStore(
+  options: {
+    dataService: DataService;
+    globalAppRegistry: AppRegistry;
+    localAppRegistry: AppRegistry;
+    query?: unknown;
+    aggregation?: unknown;
+    editViewName?: string;
+  } & CollectionMetadata
+) {
+  const {
+    dataService,
+    globalAppRegistry,
+    localAppRegistry,
+    query,
+    aggregation,
+    editViewName,
+    ...collectionMetadata
+  } = options;
+
+  const instance = (
+    globalAppRegistry.getStore('App.InstanceStore') as
+      | {
+          getInstance(): MongoDBInstance;
+        }
+      | undefined
+  )?.getInstance();
+
+  if (!instance) {
+    throw new Error('Expected to get instance from App.InstanceStore');
+  }
+
+  const configureFieldStore = globalAppRegistry.getStore('Field.Store') as (
+    ...args: any
+  ) => void | undefined; // our handcrafted d.ts file doesn't match the actual code
```

Comment:
I am not addressing it in the app-registry d.ts because in 99% cases the types we have there are correct, this is literally the only exception I could find